### PR TITLE
[bitnami/drupal] New major version

### PR DESCRIPTION
**Pull request by juan131**
Signed-off-by: juan131 <juanariza@vmware.com>

**Description of the change**

This PR bumps Drupal major version since its dependency (MariaDB) bumped the major version including breaking changes. More info at https://github.com/bitnami/charts/pull/3849

**Benefits**

- Standardization.
- Reduces technical debt.

**Benefits**

**Possible drawbacks**

It breaks backwards compatibility.

**Applicable issues**

None

**Additional information**

**Checklist**

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

**Review by marcosbc — APPROVED** [review]
Awesome, thanks

File: bitnami/drupal/README.md [hunk 1]
```diff
@@ -146,25 +146,20 @@ The following table lists the configurable parameters of the Drupal chart and th
...
-| `mariadb.master.persistence.existingClaim` | Enable persistence using an existing PVC | `nil`                                          |
-| `mariadb.master.persistence.storageClass`  | PVC Storage Class                        | `nil` (uses alpha storage class annotation)    |
-| `mariadb.master.persistence.hostPath`      | Host mount path for MariaDB volume       | `nil` (will not mount to a host path)          |
-| `externalDatabase.user`                    | Existing username in the external db     | `bn_drupal`                                    |
-| `externalDatabase.password`                | Password for the above username          | `nil`                                          |
-| `externalDatabase.database`                | Name of the existing database            | `bitnami_drupal`                               |
-| `externalDatabase.host`                    | Host of the existing database            | `nil`                                          |
-| `externalDatabase.port`                    | Port of the existing database            | `3306`                                         |
```
Comment (dani8art) [inline bitnami/drupal/README.md:167]: These lines shouldn't be removed

File: bitnami/drupal/README.md [hunk 2]
```diff
@@ -269,6 +264,46 @@ $ helm install my-release --set persistence.existingClaim=PVC_NAME bitnami/drupa
 
 ## Upgrading
 
+### To 9.0.0
+
```
Comment (dani8art) [inline bitnami/drupal/README.md:268]: Add the WARNING suggested 

```
Note: please create a backup of your database before running any of those actions. The steps below would be only valid if your application (e.g. any plugins or custom code) is compatible with MariaDB 10.5.x
```

File: bitnami/drupal/templates/NOTES.txt [hunk 3]
```diff
@@ -43,25 +43,18 @@
...
 {{- if not .Values.existingSecret -}}
-  {{- $requiredDrupalPassword := dict "valueKey" "drupalPassword" "secret" $secretName "field" "drupal-password" "context" $ -}}
-  {{- $requiredDrupalPasswordError := include "common.validations.values.single.empty" $requiredDrupalPassword -}}
-  {{- $passwordErrors =  append $passwordErrors $requiredDrupalPasswordError -}}
+    {{- $secretName := include "drupal.secretName" . -}}
+    {{- $requiredDrupalPassword := dict "valueKey" "drupalPassword" "secret" $secretName "field" "drupal-password" "context" $ -}}
+    {{- $requiredDrupalPasswordError := include "common.validations.values.single.empty" $requiredDrupalPassword -}}
+    {{- $passwordValidationErrors = append $passwordValidationErrors $requiredDrupalPasswordError -}}
 {{- end -}}
+{{- if .Values.mariadb.enabled }}
```
Comment (dani8art) [inline bitnami/drupal/templates/NOTES.txt:54]: This `if` is already checked in `"common.validations.values.mariadb.passwords"`

**Review by dani8art — CHANGES_REQUESTED** [review]
Thanks for the PR, please take a look at my comments

**Comment by dani8art** [comment]
**Upgrade using `mariadb.enabled=true` works fine**

```
$ helm install drupal bitnami/drupal
```

```
$ helm upgrade drupal bitnami/drupal 
Error: UPGRADE FAILED: template: drupal/templates/NOTES.txt:59:4: executing "drupal/templates/NOTES.txt" at <include "common.errors.upgrade.passwords.empty" (dict "validationErrors" $passwordValidationErrors "context" $)>: error calling include: template: drupal/charts/common/templates/_errors.tpl:18:48: executing "common.errors.upgrade.passwords.empty" at <fail>: error calling fail: 
PASSWORDS ERROR: you must provide your current passwords when upgrade the release
    'drupalPassword' must not be empty, please add '--set drupalPassword=$DRUPAL_PASSWORD' to the command. To get the current value:

        export DRUPAL_PASSWORD=$(kubectl get secret --namespace default drupal -o jsonpath="{.data.drupal-password}" | base64 --decode)

    'mariadb.auth.rootPassword' must not be empty, please add '--set mariadb.auth.rootPassword=$MARIADB_ROOT_PASSWORD' to the command. To get the current value:

        export MARIADB_ROOT_PASSWORD=$(kubectl get secret --namespace default drupal-mariadb -o jsonpath="{.data.mariadb-root-password}" | base64 --decode)

    'mariadb.auth.password' must not be empty, please add '--set mariadb.auth.password=$MARIADB_PASSWORD' to the command. To get the current value:

        export MARIADB_PASSWORD=$(kubectl get secret --namespace default drupal-mariadb -o jsonpath="{.data.mariadb-password}" | base64 --decode)
```

```
$ helm upgrade drupal bitnami/drupal --set drupalPassword=$DRUPAL_PASSWORD --set mariadb.auth.rootPassword=$MARIADB_ROOT_PASSWORD --set mariadb.auth.password=$MARIADB_PASSWORD
```

**Upgrade using `mariadb.enabled=false` and `externalDatabase` works fine**

```
$ helm install drupal bitnami/drupal --set mariadb.enabled=false
```

```
$ helm upgrade drupal bitnami/drupal --set mariadb.enabled=false
Error: UPGRADE FAILED: template: drupal/templates/NOTES.txt:59:4: executing "drupal/templates/NOTES.txt" at <include "common.errors.upgrade.passwords.empty" (dict "validationErrors" $passwordValidationErrors "context" $)>: error calling include: template: drupal/charts/common/templates/_errors.tpl:18:48: executing "common.errors.upgrade.passwords.empty" at <fail>: error calling fail: 
PASSWORDS ERROR: you must provide your current passwords when upgrade the release
    'drupalPassword' must not be empty, please add '--set drupalPassword=$DRUPAL_PASSWORD' to the command. To get the current value:

        export DRUPAL_PASSWORD=$(kubectl get secret --namespace default drupal -o jsonpath="{.data.drupal-password}" | base64 --decode)
```

```
$ helm upgrade drupal bitnami/drupal --set mariadb.enabled=false --set drupalPassword=$DRUPAL_PASSWORD
```

**Upgrade process from the previous chart to this new major works fine**

File: bitnami/drupal/README.md [hunk 4]
```diff
@@ -52,140 +52,140 @@ The following table lists the configurable parameters of the Drupal chart and th
...
+| `mariadb.primary.persistence.existingClaim` | Name of an existing `PersistentVolumeClaim` for MariaDB primary replicas                                              | `nil`                                                        |
+| `mariadb.primary.persistence.accessModes`   | Database Persistent Volume Access Modes                                                                               | `[ReadWriteOnce]`                                            |
+| `mariadb.primary.persistence.size`          | Database Persistent Volume Size                                                                                       | `8Gi`                                                        |
+| `mariadb.primary.persistence.storageClass`  | PVC Storage Class                                                                                                     | `nil` (uses alpha storage class annotation)                  |
+| `mariadb.primary.persistence.storageClass`  | MariaDB primary persistent volume storage Class                                                                       | `nil`                                                        |
```
Comment (FraPazGal) [inline bitnami/drupal/README.md:162]: This should be `mariadb.master.persistence.hostPath`. It looks like a typo as there is two entries for `storageClass`.

Comment (dani8art) [reply bitnami/drupal/README.md:162]: yes it should be `mariadb.primary.persistence.hostPath`, thanks!

File: bitnami/drupal/templates/NOTES.txt [hunk 5]
```diff
@@ -43,25 +43,18 @@
   echo Username: {{ .Values.drupalUsername }}
   echo Password: $(kubectl get secret --namespace {{ .Release.Namespace }} {{ template "drupal.secretName" . }} -o jsonpath="{.data.drupal-password}" | base64 --decode)
 
-{{- include "drupal.checkRollingTags" . }}
-
-{{- $passwordErrors := list -}}
-{{- $secretName := include "drupal.secretName" . -}}
-
+{{- include "common.warnings.rollingTag" .Values.image }}
```
Comment (marcosbc) [inline bitnami/drupal/templates/NOTES.txt:46]: Is this missing?
```{{- include "common.warnings.rollingTag" .Values.metrics.image -}}```

Comment (dani8art) [reply bitnami/drupal/templates/NOTES.txt:46]: rigth! 👍

**Review by FraPazGal — APPROVED** [review]
LGTM

**Review by dani8art — APPROVED** [review]

**Pull request merged**